### PR TITLE
32 selection sort subtask of 12

### PR DESCRIPTION
**Pull request by BavlyRemon**
Created the Selection sort subtask along with driver code to test and validate its functionality by running a few test cases and printing the outputs.

### CODE REVIEW

File: app/src/main/java/com/scrumptious/algorithmvisualizer/SelectionSortSequenceGen.kt
Comment (OmarElfouly) [inline app/src/main/java/com/scrumptious/algorithmvisualizer/SelectionSortSequenceGen.kt:1]: Try adding some the result of some of your tests to the bottom of the file as comments

File: app/src/main/java/com/scrumptious/algorithmvisualizer/SelectionSortSequenceGen.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        steps.add(arr.copyOf()) // Store the current state in the steps array
+        result.add(Pair(steps.last().copyOf(), swaps.toTypedArray()))
+    }
+
+    return result.toTypedArray()
+}
+
+fun main() {
```
Comment (OmarElfouly) [inline app/src/main/java/com/scrumptious/algorithmvisualizer/SelectionSortSequenceGen.kt:37]: Comment this out please.

**Review by OmarElfouly — CHANGES_REQUESTED** [review]
Shoul be good after documenting your test cases

File: app/src/main/java/com/scrumptious/algorithmvisualizer/SelectionSortSequenceGen.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package com.scrumptious.algorithmvisualizer
+
+
+class SelectionSortSequenceGen {
+fun selectionSortSequenceGenerator(arr: Array<Int>): Array<Pair<Array<Int>, Array<Int>>> {
```
Comment (OmarElfouly) [inline app/src/main/java/com/scrumptious/algorithmvisualizer/SelectionSortSequenceGen.kt:5]: This needs the result attribute in order to be consistent with the Class Diagram

**Review by OmarElfouly — CHANGES_REQUESTED** [review]

**Comment by iRustom** [comment]
Looks good!

**Pull request merged**